buildkite/client/pipeline.go

```diff
@@ -93,26 +109,66 @@ func (c *Client) CreatePipeline(pipeline *Pipeline) (*Pipeline, error) {
 	return &result, nil
 }
 
-// createYAMLPipeline will create the pipeline but only set the required fields
-// filled with stubs. After creation, UpdatePipeline will be used to set the rest
-// of the fields via the REST API
-func (c *Client) createYAMLPipeline(pipeline *Pipeline) (*Pipeline, error) {
-	created, err := c.CreatePipeline(&Pipeline{
-		Name:       pipeline.Name,
-		Repository: pipeline.Repository,
-		TeamUUIDs:  pipeline.TeamUUIDs,
-		Steps: []Step{
-			{
-				Type:    "script",
-				Name:    "Script",
-				Command: "command.sh",
-			},
-		},
-	})
+// createPipelineGraphQl will create the pipeline but only set the required fields
+// after creation, UpdatePipeline will be used to set the rest of the fields via
+// the REST API
+func (c *Client) createPipelineGraphQl(pipeline *Pipeline) (*Pipeline, error) {
+	req := graphql.NewRequest(`
+mutation PipelineCreateRequest($pipelineCreateInput: PipelineCreateInput!) {
+  pipelineCreate(input: $pipelineCreateInput) {
+    pipeline {
+      slug
+    }
+  }
+}`)
+
+	orgID, err := c.GetOrganizationId(c.orgSlug)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create an empty pipeline %s", pipeline.Name)
+		return nil, err
+	}
+
+	pci := map[string]interface{}{
+		"organizationId": orgID,
+		"name":           pipeline.Name,
+		"repository": map[string]string{
+			"url": pipeline.Repository,
+		},
+		"steps": map[string]string{
+			"yaml": pipeline.Configuration,
+		},
+	}
+
+	if len(pipeline.TeamUUIDs) != 0 {
+		var teamIDs []map[string]string
+		// Converting a slice of team UUIDs into the slice of maps since GraphQL API expects this data in this shape.
+		// We grant "MANAGE_BUILD_AND_READ" access level to _initial_ teams-owners. This can be changed later via
+		// "buildkite_team_pipeline" terraform resource.
+		for _, t := range pipeline.TeamUUIDs {
```

Comment:
Is it possible for people to specify duplicated TeamUUIDs? Should we handle that gracefully or error out? Is that something that the Buildkite APIs already handles?

Reply:
We handle it already by using https://github.com/hashicorp/terraform-plugin-sdk/blob/master/helper/schema/set.go#L50 🙂